src/lib.rs

```diff
@@ -265,14 +263,14 @@ impl Context {
         }
     }
 
-    pub fn socket(&mut self, socket_type: SocketType) -> Result<Socket, Error> {
+    pub fn socket<'a>(&'a self, socket_type: SocketType) -> Result<Socket<'a>, Error> {
```

Comment:
This has to be &self now, since the Socket has to live as long as the pointer, which would remove the ability to alias the Context, making the API incredibly restrictive. See also #64